### PR TITLE
[Commands] Auto Register Command Processors

### PR DESCRIPTION
**Pull request by OoLunar**
# Summary
Users now no longer have to explicitly register all types of command processors. Instead a new `RegisterDefaultCommandProcessors` configuration option has been added. All command processors that do not require any extra configuration will be added by default. At the time of writing, those command processors are `TextCommandProcessor`, `SlashCommandProcessor`, `UserCommandProcessor`, `MessageCommandProcessor`.

When `RegisterDefaultCommandProcessors`, `RefreshAsync` will attempt to add any missing default command processors. We do this after the user registers their processors to prevent creating the processors multiple times.

# Notes
Tested. Closes #1762

### CODE REVIEW

**Comment by Plerx2493** [comment]
How does it work when i only want to provide slashcommands only or text only?

**Comment by OoLunar** [comment]
> How does it work when i only want to provide slashcommands only or text only?

You set `RegisterDefaultCommandProcessors` to false in your `CommandsConfiguration` and then register them explicitly via `CommandsExtension.AddProcessorsAsync`

**Comment by Plerx2493** [comment]
Oh i missed that config, nice

File: DSharpPlus.Commands/CommandsExtension.cs [hunk 1]
```diff
@@ -266,14 +276,20 @@ public async Task RefreshAsync()
...
+            _processors.TryAdd(typeof(UserCommandProcessor), new UserCommandProcessor());
+        }
 
         foreach (ICommandProcessor processor in this._processors.Values)
         {
             await processor.ConfigureAsync(this);
         }
     }
 
-    // TODO: Disposable pattern for processors
```
Comment (InFTord) [inline DSharpPlus.Commands/CommandsExtension.cs:276]: wires emoji

Comment (Plerx2493) [reply DSharpPlus.Commands/CommandsExtension.cs:276]: Was discussed in libdev

Comment (OoLunar) [reply DSharpPlus.Commands/CommandsExtension.cs:276]: See #lib-development.

**Review by akiraveliara — APPROVED** [review]

**Pull request merged**